### PR TITLE
feat: add support for markdown linting

### PR DESCRIPTION
**Pull request by smira**
This does some refactoring to the automatic project builder,
incorporates new language "markdown" and required building blocks
for markdown-lint.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 Kres is a tool to automate generation of build instructions based on project structure.
 
-At the moment only Go projects are supported. Kres is opinionated, that's by design.
+At the moment only Go projects are supported.
+Kres is opinionated, that's by design.
```
Comment (andrewrynhard) [inline README.md:6]: I see the linter is working :smile:

Comment (smira) [reply README.md:6]: I had to have a test for the linter :)

File: hack/git-chglog/CHANGELOG.tpl.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <!-- THIS FILE WAS AUTOMATICALLY GENERATED, PLEASE DO NOT EDIT. -->
 <!--  -->
-<!-- Generated on 2020-08-18T18:29:05Z by kres 215d1f3-dirty. -->
+<!-- Generated on 2020-08-19T15:40:23Z by kres 29f29d8-dirty. -->
```
Comment (smira) [inline hack/git-chglog/CHANGELOG.tpl.md:3]: I fixed this files being re-generated each time below in `files.go`

**Pull request merged**